src/Paprika/Utils/Xor8.cs

```diff
@@ -147,7 +147,7 @@ public Xor8(IReadOnlyCollection<ulong> keys)
 
         _seed = seed;
 
-        var fp = new byte[arrayLength];
+        IntPtr fp = Marshal.AllocHGlobal(arrayLength);
```

Comment:
Better to use `ArrayPool`? Here the allocations are still happening just unmanaged which is a harder heap to clean from fragmentation (making allocations more expesnive and memory use increase over time)

Reply:
Yeah, makes sense. ArrayPool will give contiguous memory allocation, and that in turn will be more efficient as compared to unmanaged buffer. Thanks for pointing that out.